src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java

```diff
@@ -39,85 +40,148 @@ public ModelingSubmissionService(ModelingSubmissionRepository modelingSubmission
+            try {
+                // create new result
+                resultRepository.insertIfNonExisting(participation.getId());
+                Optional<Result> newResult = resultRepository.findFirstByParticipationIdAndRatedOrderByCompletionDateDesc(participation.getId(), false);
+                if (newResult.isPresent()) {
+                    // the insert of the new result was successful
+                    // initialize the attributes of the result
+                    result = initializeResult(participation, newResult.get());
+                } else {
+                    result = initializeResult(participation, null);
+                }
+            } catch (Exception e) {
+                throw new ConflictException("Conflict exception", "Tried to call createModelingSubmission() more than once for the same participation");
+            }
         } else {
+            // an unrated result was found for the participation
             result = optionalResult.get();
         }
 
-        result.setCompletionDate(ZonedDateTime.now());
-        modelingSubmission.setSubmissionDate(result.getCompletionDate());
-
-        if (modelingSubmission.getModel() != null && !modelingSubmission.getModel().isEmpty()) {
-            jsonModelRepository.writeModel(modelingExercise.getId(), participation.getStudent().getId(), modelingSubmission.getId(), modelingSubmission.getModel());
+        if (result.getId() == null) {
+            // there is no existing result and new result could not be created
+            return null;
         }
 
+        // update submission properties
+        modelingSubmission.setSubmissionDate(ZonedDateTime.now());
+        modelingSubmission.setType(SubmissionType.MANUAL);
+
         result.setSubmission(modelingSubmission);
         modelingSubmission.setResult(result);
         modelingSubmissionRepository.save(modelingSubmission);
         resultRepository.save(result);
 
+        if (modelingSubmission.getModel() != null && !modelingSubmission.getModel().isEmpty()) {
+            jsonModelRepository.writeModel(modelingExercise.getId(), user.getId(), modelingSubmission.getId(), modelingSubmission.getModel());
+        }
+
         if (modelingSubmission.isSubmitted()) {
             submit(modelingSubmission, modelingExercise);
+        } else if (modelingExercise.getDueDate() != null && !modelingExercise.isEnded()) {
+            // save submission to HashMap if exercise not ended yet
+            AutomaticSubmissionService.updateSubmission(modelingExercise.getId(), user.getLogin(), modelingSubmission);
         }
 
         return result;
     }
 
+    /**
+     * Adds a model to compass service to include it in the automatic grading process.
+     *
+     * @param modelingSubmission    the submission which contains the model
+     * @param modelingExercise      the exercise the model belongs to
+     */
     public void submit(ModelingSubmission modelingSubmission, ModelingExercise modelingExercise) {
```

Comment:
The comment does not really fit to the method name. Is this really a submit functionality? If yes, would we not need to remove the submission from the scheduling hash map? If not, please rename the method appropriately